### ISSUE TITLE
Listorders CLI output - peerpubkey

### ISSUE THREAD

**Issue by sangaman** [ExchangeUnion/xud#643]
After #600, our `getorders` CLI output looks like this:

![image](https://user-images.githubusercontent.com/3440835/48082856-9e906100-e1c1-11e8-82f9-45a06ad67666.png)

I'm thinking it would be nice to show which peer the owner belongs to. We could probably repurpose the `Own Order` column for this. Maybe change the label to something like "Owner" or "Peer" and print the node's peer pub key for peer orders, and "Own" or "Local" for own orders. It may even be worth taking it a step further and printing the peer's alias if one exists - however we don't have that data currently available from the `getorders` rpc call. If we think it's worth it, we could make cli command also call `listpeers` to get all the aliases and then map them to the pubkeys.

Thoughts @erkarl @kilrau @offerm?

**Comment by offerm** [comment]
@sangaman Initially, I thought that this is a great idea.  Problems that I see:
- do we have xud node alias support? 
- will these aliases be used by the peers? can you trust the alias? will it give valuable information
- do you have room to show it?
- If you show host name if we have it
- not sure if IP address provides value.

After thinking about it more, and having a close look at the table, I think that:
a. we need to sort the buy/sell orders according to prices.
b. We need to aggregate the capacity available at each price level (rounded). For example, in the table above we have 2 BUY orders of at price of 0.87.  These should be combined.
c. once you combine and sort properly you will be able to see the market depth. If we will not aggregate but have a lot of orders this display will not be usable. 
d. I would consider to have this as a new cli `getmarketdept` and leave `getorders` as today.
e. to be considered - aggregate the buy/sell orders into n (say 8) buy buckets and n sell bucket and presenting them over the price line. So sell orders will come first (higher prices) and buy order will come below it (normal view of order book).

With a-e, I don't see a need for `owner` column.

**Comment by erkarl** [comment]
I'm really hesitant in regards to showing aliases. If they are like lnd's aliases they can't be trusted and just cause confusion.

a-b. Yes, definitely. That's something I left out of the original PR intentionally.
c-e. I imagine `getorders` with the current implementation will not be usuable in any production like environment because of too many orders. Maybe the default of `getorders` should be to return the market depth/aggregated results.

**Comment by sangaman** [comment]
Ok, maybe leave the alias aside for now, it's true that it's not implemented (but I'm thinking it would be nice to have locally specified aliases, made a comment on that issue).

For now, what about just showing the peer pub key or `Own` or `Local` for own orders? Later it can possibly be enhanced to show aliases.

I'll create a separate issue for the other points raised since I think they are tangential to this one.

**Comment by sangaman** [comment]
Another thought is that if we do display the peer pub key, it need not be the whole thing. First 8-10 characters should be plenty.

**Comment by kilrau** [comment]
Agree

**Comment by kilrau** [comment]
The more I think about this one, the more it looks like a "debug" feature. Anyhow, I'd be fine with adding it if:
* we make the [market depth](https://github.com/ExchangeUnion/xud/issues/648) version the new default for `listorders` since this is how all exchanges display the order book and the only thing that makes sense in real-life usage.
* Keep the existing `listorders` as e.g. `listorderssingle` and add the peerPubKey there. Maybe as [...587D3F] last 6 chars only.

Other suggestions for naming?

**Comment by erkarl** [comment]
I agree on showing the market depth for `listorders` so it will be easier to onboard new users with a familiar experience. 

`listorderssingle` could also be `listorders --debug` or `listorders --single` or `listorders --all`.

**Comment by kilrau** [comment]
I like `listorders --debug` since we can apply the `--debug` flag to any "more detailed" version of a call.

**Comment by kilrau** [comment]
Ok with you? @sangaman

**Comment by sangaman** [comment]
I think it would make sense for market depth output to be in a separate call like `marketdepth`, and keep `listorders` in line with the actual `ListOrders` rpc call by having it list out individual orders, where I still think some sort of peer identifier makes sense.

**Comment by kilrau** [comment]
Keeping it in line with the actual rpc calls - that makes sense. But I also would want the default order book view to be in line with how exchanges currently present their order books; and that's a market depth view of orders.

Maybe the better alternative: we leave the `listorders` call as is, but change the [description to: a list of all single orders in the order book for a given pair_id, along with peer info and other details.](https://github.com/ExchangeUnion/xud/blob/ed0afb218adab3a83ccb2ac2a9e08d8b1597f40a/lib/cli/commands/listorders.ts#L94). Create a new call `orderbook [pair_id]` for marketdepth. That's also how our friends from sprkswp call it btw.

If that sounds good to everyone this issue would simply be about adding the alias (once https://github.com/ExchangeUnion/xud/issues/486 is done) with last X digits of peers pubkey to `listorders` output. I would open a new issue for the `orderbook [pair_id]`  call. Opinions? @erkarl @sangaman

**Comment by erkarl** [comment]
Recap from call: @sangaman's concern was to not have this on xud level.

**Comment by rsercano** [comment]
Is there any news on this ? I read every single comment on this one but still I think it would be great if you can finalize what has to be done guys

**Comment by kilrau** [comment]
Let's rename the column "Own Order" to "Node". If it's my own it says "Own" instead of "X", if it's from a peer it displays the Alias (not node key).

Btw how is https://github.com/ExchangeUnion/xud/issues/486 coming along? @hatmer

**Comment by hatmer** [comment]
#486 is going well, the alias generation part is complete. I'm having difficulties getting the alias to show up in the CLI though, should I perhaps be manually recompiling the protocol buffer files?

**Comment by hatmer** [comment]
Nvm I figured it out. Still adding some functionality but getting close to pushing.
